components/director/cmd/director/main.go

```diff
@@ -181,6 +181,8 @@ type config struct {
 
 	ExternalClientCertSecretName string `envconfig:"APP_EXTERNAL_CLIENT_CERT_SECRET_NAME"`
 	ExtSvcClientCertSecretName   string `envconfig:"APP_EXT_SVC_CLIENT_CERT_SECRET_NAME"`
+
+	ApplicationTemplateProductLabel string `envconfig:"APP_APPLICATION_TEMPLATE_PRODUCT_LABEL"`
```

Comment:
As far as I see this label is used only in the SelfReg flow so why don't we put it into the `SelfRegConfig`? This will prevent the need from passing the label as an argument to many functions. And as i know this label is also some kind of "distinguish label", so we can keep the convention of naming. Maybe something like `SelfRegisterAppTemplateProductLabelKey`.

Reply:
Because this label has nothing to do with the self reg flow. We use it in the SelfReg manager, yes, but that's only because the `if`s are there that check the flows. The label is part of the CLD flow and that's why it is not in the SelfReg Config